### PR TITLE
Use shootUID for randomization of backup schedule

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:
This PR fixes https://github.com/gardener/gardener-extensions/pull/337#issuecomment-537857510. And use shootUID for randomiaztion of schedule
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Comment by swapnilgm** [comment]
cc @ialidzhikov

File: pkg/webhook/controlplane/etcd.go [hunk 1]
```diff
@@ -170,7 +176,7 @@ func DetermineBackupSchedule(c *corev1.Container, cluster *extensionscontroller.
 			// Randomize the snapshot timing daily but within last hour.
 			// The 15 minutes buffer is set to snapshot upload time before actual maintainance window start.
 			snapshotWindowBegin := maintenanceTimeWindow.Begin().Add(-1, -15, 0)
-			randomMinutes := rand.Intn(59)
+			randomMinutes := int(crc32.ChecksumIEEE([]byte(shootUID)) / 59)
```
Comment (ialidzhikov) [inline pkg/webhook/controlplane/etcd.go]: Shouldn't it to be `% 59`?

```suggestion
			randomMinutes := int(crc32.ChecksumIEEE([]byte(shootUID)) % 59)
```

File: controllers/provider-alicloud/pkg/webhook/controlplanebackup/ensurer_test.go [hunk 2]
```diff
@@ -250,6 +259,43 @@ var _ = Describe("Ensurer", func() {
...
+		It("should not modify elements to same etcd-events statefulset", func() {
+			var (
+				ss = &appsv1.StatefulSet{
+					ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: v1alpha1constants.StatefulSetNameETCDEvents},
+				}
+			)
+
+			// Create mock client
+			client := mockclient.NewMockClient(ctrl)
+			//client.EXPECT().Get(context.TODO(), secretKey, &corev1.Secret{}).DoAndReturn(clientGet(secret))
```
Comment (ialidzhikov) [inline controllers/provider-alicloud/pkg/webhook/controlplanebackup/ensurer_test.go]: You can remove the commented code from each test.

Comment (swapnilgm) [reply controllers/provider-alicloud/pkg/webhook/controlplanebackup/ensurer_test.go]: Done. Thanks

**Pull request merged**